src/config/config.rs

```diff
@@ -157,36 +162,28 @@ impl Type {
   }
 }
 
-#[derive(Serialize, Deserialize, Clone, Debug, Default)]
-pub struct GraphQL {
-  pub schema: RootSchema,
-  #[serde(default)]
-  pub types: BTreeMap<String, Type>,
-  #[serde(default)]
-  pub unions: BTreeMap<String, Union>,
-}
-
-impl GraphQL {
-  pub fn merge_right(mut self, other: Self) -> Self {
-    for (name, mut other_type) in other.types {
-      if let Some(self_type) = self.types.remove(&name) {
-        other_type = self_type.merge_right(&other_type)
-      };
+fn merge_types(mut self_types: BTreeMap<String, Type>, other_types: BTreeMap<String, Type>) -> BTreeMap<String, Type> {
+  for (name, mut other_type) in other_types {
+    if let Some(self_type) = self_types.remove(&name) {
+      other_type = self_type.merge_right(&other_type)
+    };
 
-      self.types.insert(name, other_type);
-    }
+    self_types.insert(name, other_type);
+  }
+  self_types
+}
 
-    for (name, mut other_union) in other.unions {
-      if let Some(self_union) = self.unions.remove(&name) {
-        other_union = self_union.merge_right(other_union);
-      }
-      self.unions.insert(name, other_union);
+fn merge_unions(
+  mut self_unions: BTreeMap<String, Union>,
+  other_unions: BTreeMap<String, Union>,
+) -> BTreeMap<String, Union> {
+  for (name, mut other_union) in other_unions {
+    if let Some(self_union) = self_unions.remove(&name) {
+      other_union = self_union.merge_right(other_union);
     }
-
-    self.schema = self.schema.merge_right(other.schema);
-
-    self
+    self_unions.insert(name, other_union);
   }
+  self_unions
 }
```

Comment:
It'd make more sense to have these as private functions from `Config`.

Reply:
These functions are private to Config.